explorer/types.go

```diff
@@ -123,9 +123,38 @@ type SiafundOutput struct {
 	types.SiafundElement
 }
 
-// A FileContract is a types.FileContractElement with added fields for
-// resolved/valid state.
+type ContractSiacoinOutput struct {
+	ID types.SiacoinOutputID `json:"id"`
+	types.SiacoinOutput
+}
+
+// A FileContract is a storage agreement between a renter and a host. It
+// contains a bidirectional payment channel that resolves as either "valid" or
+// "missed" depending on whether a valid StorageProof is submitted for the
+// contract.
 type FileContract struct {
+	Filesize           uint64                  `json:"filesize"`
+	FileMerkleRoot     types.Hash256           `json:"fileMerkleRoot"`
+	WindowStart        uint64                  `json:"windowStart"`
+	WindowEnd          uint64                  `json:"windowEnd"`
+	Payout             types.Currency          `json:"payout"`
+	ValidProofOutputs  []ContractSiacoinOutput `json:"validProofOutputs"`
+	MissedProofOutputs []ContractSiacoinOutput `json:"missedProofOutputs"`
+	UnlockHash         types.Hash256           `json:"unlockHash"`
+	RevisionNumber     uint64                  `json:"revisionNumber"`
+}
+
+// A FileContractElement is a record of a FileContract within the state
+// accumulator.
+type FileContractElement struct {
```

Comment:
I think you can drop `FileContractElement` from the FileContract type. You should be able to replace it with just the ID and get everything necessary.

```
// A FileContract is a storage agreement between a renter and a host. It
// contains a bidirectional payment channel that resolves as either "valid" or
// "missed" depending on whether a valid StorageProof is submitted for the
// contract.
type FileContract struct {
	ID                 types.FileContractID    `json:"id"`
	Filesize           uint64                  `json:"filesize"`
	FileMerkleRoot     types.Hash256           `json:"fileMerkleRoot"`
	WindowStart        uint64                  `json:"windowStart"`
	WindowEnd          uint64                  `json:"windowEnd"`
	Payout             types.Currency          `json:"payout"`
	ValidProofOutputs  []ContractSiacoinOutput `json:"validProofOutputs"`
	MissedProofOutputs []ContractSiacoinOutput `json:"missedProofOutputs"`
	UnlockHash         types.Hash256           `json:"unlockHash"`
	RevisionNumber     uint64                  `json:"revisionNumber"`
}

// A FileContractRevision is a FileContract with extra fields for revision
// information.
type FileContractRevision struct {
	ParentID           types.FileContractID    `json:"parentID"`
	UnlockConditions   types.UnlockConditions  `json:"unlockConditions"`
	Filesize           uint64                  `json:"filesize"`
	FileMerkleRoot     types.Hash256           `json:"fileMerkleRoot"`
	WindowStart        uint64                  `json:"windowStart"`
	WindowEnd          uint64                  `json:"windowEnd"`
	ValidProofOutputs  []ContractSiacoinOutput `json:"validProofOutputs"`
	MissedProofOutputs []ContractSiacoinOutput `json:"missedProofOutputs"`
	UnlockHash         types.Hash256           `json:"unlockHash"`
	RevisionNumber     uint64                  `json:"revisionNumber"`
}
```

Reply:
How do you feel about the `StateElement`?  The `LeafIndex` is useful for retrieving the `MerkleProof`.

Reply:
leaf index and merkle proof are unnecessary for v1 file contracts